src/store/modules/app.js

```diff
@@ -73,6 +89,41 @@ const getters = {
 };
 
 const actions = {
+  checkValidKeymap(_, keymap) {
+    if (
```

Comment:
Seems about the same, with less boilerplate.

```suggestion
    return isUndefined(keymap.keyboard) || isUndefined(keymap.keymap)  || isUndefined(keymap.layout) || isUndefined(keymap.layers);
```